sandboxes/staticanalysis/staticanalyze.go

```diff
@@ -0,0 +1,235 @@
+		parserType = "custom"
+	}
+
+	parserAbsPath, err := filepath.Abs(parserPath)
+	if err != nil {
+		return "", fmt.Errorf("could not resolve relative path %s for JS parser: %v", parserPath, err)
+	}
+	if _, err = os.Stat(parserAbsPath); err != nil {
+		return "", fmt.Errorf("could not locate %s JS parser at %s: %v", parserType, parserAbsPath, err)
+	}
+
+	return parserPath, nil
+}
+
+func checkAnalyses(names []string) ([]staticanalysis.Task, error) {
+	var tasks []staticanalysis.Task
+	for _, name := range names {
+		task, ok := staticanalysis.TaskFromString(name)
+		if !ok {
+			return nil, fmt.Errorf("unrecognised static analysis task: %s", name)
+		}
+		tasks = append(tasks, task)
+	}
+	return tasks, nil
+}
+
+func printAnalyses() {
+	println("Available analyses are:")
+	for _, task := range staticanalysis.AllTasks() {
+		println("\t", task)
+	}
+}
+
+func doObfuscationDetection(extractedPackageDir string) (*obfuscation.AnalysisResult, error) {
+	jsParserPath, err := getJsParserPath()
+	if err != nil {
+		return nil, err
+	}
+
+	result := &obfuscation.AnalysisResult{
+		FileRawData:    map[string]obfuscation.RawData{},
+		FileSignals:    map[string]obfuscation.Signals{},
+		PackageSignals: obfuscation.NoSignals(),
+		ExcludedFiles:  []string{},
+	}
+	err = filepath.WalkDir(extractedPackageDir, func(path string, f fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if f.Type().IsRegular() {
+			pathInArchive := strings.TrimPrefix(path, extractedPackageDir+string(os.PathSeparator))
+			fmt.Printf("Processing file %s\n", pathInArchive)
+			rawData, err := obfuscation.CollectData(jsParserPath, path, "", false)
+			if err != nil {
+				log.Error("Error parsing file", "filename", pathInArchive, "error", err)
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else if rawData == nil {
+				// syntax error - could not parse file
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else {
+				// rawData != nil, err == nil
+				result.FileRawData[f.Name()] = *rawData
+				signals := obfuscation.ComputeSignals(*rawData)
+				result.FileSignals[f.Name()] = obfuscation.RemoveNaNs(signals)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error while walking package files: %v", err)
+	}
+
+	return result, nil
+}
+
+func cleanupTmpDir(dir string) {
+	err := os.RemoveAll(dir)
+	if err != nil {
+		fmt.Printf("Failed to remove temp work dir: %v\n", err)
+	}
+}
+
+func makeWorkDirs() (tmpDir, archiveDir, extractDir string, err error) {
+	if tmpDir, err = os.MkdirTemp("", "package-analysis-staticanalyze"); err != nil {
+		return "", "", "", err
+	}
+
+	if archiveDir, err = os.MkdirTemp(tmpDir, "archive"); err != nil {
+		cleanupTmpDir(tmpDir)
+		return "", "", "", err
+	}
+
+	if extractDir, err = os.MkdirTemp(tmpDir, "extracted"); err != nil {
+		cleanupTmpDir(tmpDir)
+		return "", "", "", err
+	}
+
+	return
+}
+
+func mainReturningError() (err error) {
```

Comment:
nit: no need to have the "ReturningError" part in the function name.

Reply:
The `main` function isn't allowed to return anything though - I wanted to have a function where I could return an error and have the actual `main` function call `os.Exit()` if the error is not `nil`. Calling `os.Exit()` directly in this function means any `defer`-ed functions will be skipped.

Perhaps I could call this function something else?

Reply:
Yeah, my comment was more on this being a strange way to name this. I.e. it's describing a workaround rather than what it's actualy doing. 

Something like: "runStaticAnalysisCLI" would be a better name for this.

Reply:
How about `run()`?

https://pace.dev/blog/2020/02/12/why-you-shouldnt-use-func-main-in-golang-by-mat-ryer.html#solution-a-mini-abstraction-4

Reply:
Yes run is also good.